packages/core/src/shared/telemetry/activation.ts

```diff
@@ -49,12 +50,23 @@ export async function activate(
     globals.telemetry = await DefaultTelemetryService.create(extensionContext, awsContext, getComputeRegion())
 
     try {
-        globals.telemetry.telemetryEnabled = config.isEnabled()
+        await globals.telemetry.setTelemetryEnabled(config.isEnabled())
 
         extensionContext.subscriptions.push(
-            (isAmazonQ() ? config.amazonQConfig : config.toolkitConfig).onDidChange(event => {
+            (isAmazonQ() ? config.amazonQConfig : config.toolkitConfig).onDidChange(async event => {
                 if (event.key === 'telemetry') {
-                    globals.telemetry.telemetryEnabled = config.isEnabled()
+                    const val = config.isEnabled()
+                    if (!val) {
+                        // Record 'disabled' right before its turned off, so we can send this + the batch we have already.
+                        telemetry.ui_click.emit({ elementId: 'aws_disableTelemetry' })
+                    }
+
+                    await globals.telemetry.setTelemetryEnabled(val)
+
+                    if (val) {
+                        // Record 'enabled' after its turned on, otherwise this is ignored.
+                        telemetry.ui_click.emit({ elementId: 'aws_enableTelemetry' })
```

Comment:
there is an `enabled` field that could be set: https://github.com/aws/aws-toolkit-common/blob/9fdefa93c1741bf2cc3046a01f70fd7cc3eed540/telemetry/definitions/commonDefinitions.json#L1064-L1068

then we can have a single metric (or in this case, "ui_click element") that has multiple states. Instead of representing every state with a new "metric"